eisbuk-admin/src/components/atoms/IntervalCard/__tests__/IntervalCard.test.tsx

```diff
@@ -1,16 +1,75 @@
 import React from "react";
-import { render } from "@testing-library/react";
+import {cleanup, screen render } from "@testing-library/react";
 
 import IntervalCard from "../IntervalCard";
+import * as bookingActions from "@/store/actions/bookingOperations";
+import { bookedSlot, nonBookedSlot } from "../__testData__/dummyData";
 
-import { baseProps } from "../__testData__/dummyData";
+import { Customer, Slot } from "eisbuk-shared";
+import { SlotInterface, SlotInterval } from "@/types/temp";
+
+
+const mockDispatch = jest.fn();
+
+jest.mock("react-redux", () => ({
+  ...jest.requireActual("react-redux"),
+  useDispatch: () => mockDispatch,
+  /** We're mocking this not to fail tests including SlotForm (this will be returning newSlotTime as null) @TODO remove in the future */
+  useSelector: () => null,
+}));
+
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useParams: () => ({ secretKey: "secret_key" }),
+}));
+
+/** @TODO remove this when the i18next is instantiated with tests */
+jest.mock("i18next", () => ({
+  ...jest.requireActual("i18next"),
+  /** We're mocking this not to fail certain tests depending on this, but we're not testing the i18n values, so this is ok for now @TODO fix i18n in tests */
+  t: () => "",
+}));
 
 describe("Interval Card ->", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+    cleanup();
+  });
+
+  // describe
+  test("should render properly", () => {
+    render(<IntervalCard {...bookedSlot} />);
+  });
   describe("Test button functionality ->", () => {
-    test("delete this test when you add your own", () => {
-      render(<IntervalCard {...baseProps} />);
+    const mockBookInterval = jest.spyOn(bookingActions, "bookInterval");
+    const mockCancelBooking = jest.spyOn(bookingActions, "cancelBooking");
```

Comment:
Here you're spying on redux dispatches, while in our component we're receiving functions as props, so to test them out, you simply create two mock functions, pass them to the component and test them being called, i.e.
```jsx
const mockBookInterval = jest.fn()
const mockCancelBooking = jest.fn()

render(<IntervalCard booked bookInterval={mockBookInterval} cancelBooking={cancelBooking} />
/* some test actions */
expect(mockBookInterval).toHaveBeenCalled()
```